rtil/src/threads/ue/rebo/livesplit.rs

```diff
@@ -97,4 +126,42 @@ impl Run {
     pub fn attempt_count() -> u32 {
         LIVESPLIT_STATE.lock().unwrap().timer.run().clone().attempt_count()
     }
+    pub fn save_splits(user_input: &str) -> Result<(), SplitsSaveError> {
+        let state = LIVESPLIT_STATE.lock().unwrap();
+        let valid_splits_paths = VALID_SPLITS_PATHS.lock().unwrap();
+        let path = Path::new(user_input);
+        let (path, path_display) = match (path.is_relative(), valid_splits_paths.contains(path)) {
+            (true, _) => {
+                let sanitized = sanitize(user_input);
+                (LiveSplit::splits_path().join(&sanitized), sanitized)
+            },
+            (false, true) => (path.to_owned(), user_input.to_owned()),
+            (false, false) => return Err(SplitsSaveError::DisallowedFilePath(user_input.to_string())),
+        };
+        let file = File::create(&path)
+            .map_err(|e| SplitsSaveError::CreationFailed(path_display.clone(), e.to_string()))?;
+        save_timer(&state.timer, IoWrite(BufWriter::new(file)))
+            .map_err(|e| SplitsSaveError::SaveFailed(path_display.clone(), e.to_string()))?;
+        Ok(())
+    }
+    pub fn load_splits(user_input: &str) -> Result<(), SplitsLoadError> {
+        let path = Path::new(user_input);
+        let (path, path_display) = match path.is_relative() {
+            true => (LiveSplit::splits_path().join(user_input), LiveSplit::splits_path().
+                join(user_input)
+                .to_str().expect("Could not convert to &str")
+                .to_owned()),
+            false => (path.to_path_buf(), path
+                .to_str().expect("Could not convert to &str")
+                .to_owned()),
```

Comment:
For multiline tuples, prefer this syntax:
```suggestion
            true => (
                LiveSplit::splits_path().join(user_input),
                LiveSplit::splits_path().join(user_input)
                    .to_str().expect("Could not convert to &str").to_owned(),
            ),
            false => (
                path.to_path_buf(),
                path.to_str().expect("Could not convert to &str").to_owned(),
            ),
```